# <center>Tensorflow 学习笔记</center>

## 安装
anaconda是python的发行版，里面集成了常用的python库，如numpy，scipy等，并且有conda对包进行更新和管理，可以说对新手非常友好。

以windows平台下的安装GPU版本为例：

1. 建立虚拟环境: 我们建立了一个名称为`tensorflow`的虚拟环境，其中python的版本号是3.6

  >   `conda create --name  tensorflow python=3.6   `
     
2. 激活虚拟环境

  >  `activate tensorflow`
    
3. 安装GPU版本的tensorflow(注意：前提是你的GPU要支持并行运算，否则请安装CPU版本)。 根据最新消息，这条命名执行时会自动安装CUDA以及显卡相关的驱动等。
  >  `conda install -c anaconda tensorflow-gpu    `
   
   
4. 检测是否安装成功
  > ``` import tensorflow as tf
  tf.__version__    ``` 
  
  

In [1]:
import tensorflow as tf
tf.__version__

'1.12.0'

能够正确返回结果，说明我们已经安装成功。

可以看到，当前tensorflow版本是1.12版，估计不久后，正式的tensorflow2.0版将要发布。

我们还可以通过如下命令查看设备
 
```
sess = tf.Session()) #创建了一个会话

```
![GPU](./image/xianshi.png)

可以看到，我的笔记本的GPU为GTX960,计算能力评分为5，而tensorflow的官方建议要大于3.5，可以说勉强够用。

## 第一个程序
在命令行中输入`activtae tensorflow`激活conda虚拟环境，再输入`jupyter notebook`，后，浏览器就会打开jupyter dashboard，新建一个ipynb文件，我们就进行演示了。输入以下程序：

In [21]:
import tensorflow as tf  #载入tensorflow

a = tf.constant(1)
b = tf.constant(2) # 创建两个常数

c = a + b  # 创建operation

sess = tf.Session() # 创建会话
 
print(sess.run(c))  #打印计算结果

3


<tf.Tensor 'hah_1:0' shape=() dtype=int32>

module